src/test/testUtil.ts

```diff
@@ -146,46 +146,69 @@ export function getMetrics<K extends MetricName>(
 }
 
 /**
- * Finds the emitted telemetry metric with the given name, then checks if the metadata fields
- * match the expected values. Defaults to checking against the first occurrence, but this
- * can be overidden.
+ * Finds the emitted telemetry metrics with the given `name`, then checks if the metadata fields
+ * match the expected values, in the (non-continguous) order specified by `expected`.
  *
- * @param eventNum The nth occurrence of the event you want to test against, defaults to the first.
+ * @param name Metric name
+ * @param expected Metric(s) shape(s) which are compared _in order_ (non-continguous) to metrics matching `name`.
  */
 export function assertTelemetry<K extends MetricName>(
     name: K,
-    expected: MetricShapes[K],
-    eventNum?: number
+    expected: MetricShapes[K] | MetricShapes[K][]
 ): void | never
 export function assertTelemetry<K extends MetricName>(
     name: K,
-    expected: MetricShapes[MetricName],
-    eventNum?: number
+    expected: MetricShapes[MetricName] | MetricShapes[MetricName][]
 ): void | never
 export function assertTelemetry<K extends MetricName>(
     name: K,
-    expected: MetricShapes[K],
-    eventNum: number = 0
```

Comment:
Eliminate `eventNum`. Now the tests can pass a list to assert a particular (non-contiguous) metrics order.